packages/app/src/entities/market/hooks/use-market-data.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react'
+import { getReserve } from 'LendingPool'
+
+const PERCENT_PRECISION = 1e4
+
+type PoolData = {
+  borrowInterestRate?: bigint
+  lendInterestRate?: bigint
+  discount?: number
+  liquidationPenalty?: number
+}
+
+export function useMarketData(tokenAddress: string): PoolData & {
+  percentMultiplier: number
+} {
+  const [{ borrowInterestRate, lendInterestRate, discount, liquidationPenalty }, setData] =
```

Comment:
Looks like you dont need destructurization here
You can return a whole object from useMarketData

Reply:
updated